src/github.com/couchbase/sync_gateway/db/changes.go

```diff
@@ -304,9 +305,13 @@ func (db *Database) MultiChangesFeed(chans base.Set, options ChangesOptions) (<-
 				for i, cur := range current {
 					if cur == nil && feeds[i] != nil {
 						var ok bool
-						current[i], ok = <-feeds[i]
-						if !ok {
-							feeds[i] = nil
+						// Add select to ensure we don't skip feeds that are non-closed but still
+						// processing the channel update.
+						select {
+						case current[i], ok = <-feeds[i]:
+							if !ok {
+								feeds[i] = nil
+							}
```

Comment:
After discussion, looks like this is not needed.  I removed it and retested it with -cpu 4 -race, and all tests passed.

@adamcfraser can you push a new commit with the select call removed?  (I'd do it my editor isn't configured yet)